src/crimson/os/seastore/segment_manager/zns.cc

```diff
@@ -489,18 +489,44 @@ blk_close_zone_ret blk_close_zone(
   });
 }
 
+using blk_reset_zone_ertr = crimson::errorator<
+  crimson::ct_error::input_output_error>;
+using blk_reset_zone_ret = blk_reset_zone_ertr::future<>;
+  blk_reset_zone_ret blk_reset_zone(
```

Comment:
blk_open/close/finish/reset_zone have very similar implementations.

Let's extract a shared method for them to reduce the repeating logics.

Reply:
It is a good idea, which has crossed my mind also.
Is it ok to change this in next patch set, where I am planning some more cleaning and re-organising the zns code ?

Reply:
Yeah, it helps to better understand the code and less error-prone.